test/runtime_tests.cc

```diff
@@ -763,6 +763,39 @@ namespace detail {
 		REQUIRE_FALSE(bm.has_active_buffers());
 	}
 
+	TEST_CASE("buffer_manager allows buffers deallocation", "[buffer_manager][dealloc]") {
+		celerity::distr_queue q;
+		buffer_id b_id;
+		auto& bm = runtime::get_instance().get_buffer_manager();
+		auto& tm = runtime::get_instance().get_task_manager();
+		int new_horizon_step = 2;
+		tm.set_horizon_step(new_horizon_step);
+		{
+			celerity::buffer<int, 1> b(celerity::range<1>(128));
+			b_id = celerity::detail::get_buffer_id(b);
+			q.submit([=](celerity::handler& cgh) {
+				celerity::accessor a{b, cgh, celerity::access::all(), celerity::write_only};
+				cgh.parallel_for<class UKN(i)>(b.get_range(), [=](celerity::item<1> it) {});
+			});
+			REQUIRE(bm.has_buffer(b_id));
+		}
+		celerity::buffer<int, 1> c(celerity::range<1>(128));
+		// we need horizon_step_size * 3 + 1 tasks to generate the third horizon,
+		// and one extra task to trigger the clean_up process
+		for(int i = 0; i < (new_horizon_step * 3 + 2); i++) {
+			q.submit([=](celerity::handler& cgh) {
+				celerity::accessor a{c, cgh, celerity::access::all(), celerity::write_only};
+				cgh.parallel_for<class UKN(i)>(c.get_range(), [=](celerity::item<1>) {});
+			});
+			q.slow_full_sync();
```

Comment:
It should be enough for this to happen once after the loop.

Reply:
the `slow_full_sync()` needs to be inside the `for` otherwise there is a race between the prepass and the executor informing the tdag of executed horizons. for more info pls reach @psalz.
I will add a comment on the test explaining this.

Reply:
You're right! The explanation is the same as in the `tasks behind the deletion horizon are deleted` runtime test: At least the second-to-last task must be awaited explicitly so that the last horizon is executed, otherwise creation of the final task does not trigger a cleanup.